src/machine.rs

```diff
@@ -190,7 +190,13 @@ impl<'m> Machine<'m> {
             cmd = cmd.arg(daemonize_arg);
         }
         trace!("{vm_id}: Running command: {:?}", cmd);
-        let child = cmd.spawn()?;
+        let mut child = cmd.spawn()?;
+        let pid = if let Some(id) = child.id() {
+            Ok(id)
+        } else {
+            let exit_status = child.wait().await?;
+            Err(Error::ProcessExitedEarly { exit_status })
+        }?;
```

Comment:
I think a match would be better here:

```suggestion
        let pid = match child.id() {
            Some(id) => id,
            None => {
                let exit_status = child.wait().await?;
                return Err(Error::ProcessExitedEarly { exit_status });
        };
```

No?